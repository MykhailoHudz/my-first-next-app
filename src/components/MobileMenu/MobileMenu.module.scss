@import "../../app/styles//mixins/breakpoints.scss";
@import "../../app/styles/constants.scss";

.mobile__navigation {
  position: relative;
  @include breakpoint(tablet) {
    display: none;
  }

  .menu_btn {
    border: 1px solid $accent-red-color;
    border-radius: 15px;
    padding: 5px 15px;
  }
  .list {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 3px;

    padding: 10px;
    background-color: $main-background-color;
    border: 5px solid $white-color;
    border-radius: 20px;
    box-shadow: $shadow-color 0px 30px 90px;

    position: absolute;
    top: 35px;
    left: -25%;
  }
}
