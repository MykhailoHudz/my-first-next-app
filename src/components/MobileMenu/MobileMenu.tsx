"use client";
import Link from "next/link";
import styles from "./MobileMenu.module.scss";
import { useState } from "react";

function MobileMenu() {
  const [showNavigation, setShowNavigation] = useState(false);

  const menuToggle = () => {
    setShowNavigation((prev) => !prev);
  };

  return (
    <div className={styles.mobile__navigation}>
      <button className={styles.menu_btn} type="button" onClick={menuToggle}>
        Menu
      </button>
      {showNavigation && (
        <ul className={styles.list}>
          <li>
            <Link onClick={menuToggle} className={styles.link} href={"/"}>
              Home
            </Link>
          </li>
          <li>
            <Link
              onClick={menuToggle}
              className={styles.link}
              href={"/information"}
            >
              Information
            </Link>
          </li>
          <li>
            <Link onClick={menuToggle} className={styles.link} href={"/team"}>
              Team
            </Link>
          </li>
          <li>
            <Link
              onClick={menuToggle}
              className={styles.link}
              href={"/comments"}
            >
              Comments
            </Link>
          </li>
        </ul>
      )}
    </div>
  );
}

export default MobileMenu;
