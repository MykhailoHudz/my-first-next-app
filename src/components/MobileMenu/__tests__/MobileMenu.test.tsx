import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import MobileMenu from "../MobileMenu";

// const additionalConstant = 15;
// const additionalConstant2 = 15;
// const additionalConstant3 = 15;

test("matches snapshot", () => {
    const { asFragment } = render(<MobileMenu />);

    expect(asFragment()).toMatchSnapshot();

    const button = screen.getByRole("button", { name: /menu/i });
    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
});
