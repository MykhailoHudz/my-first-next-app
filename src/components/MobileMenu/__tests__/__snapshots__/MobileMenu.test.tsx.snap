// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mobile__navigation"
  >
    <button
      class="menu_btn"
      type="button"
    >
      Menu
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="mobile__navigation"
  >
    <button
      class="menu_btn"
      type="button"
    >
      Menu
    </button>
    <ul
      class="list"
    >
      <li>
        <a
          class="link"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="link"
          href="/information"
        >
          Information
        </a>
      </li>
      <li>
        <a
          class="link"
          href="/team"
        >
          Team
        </a>
      </li>
      <li>
        <a
          class="link"
          href="/comments"
        >
          Comments
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
